widgets/common/src/hooks/useLabelInput.ts

```diff
@@ -1,31 +1,41 @@
 import { SyntheticEvent } from 'react';
-import type { DropdownProps } from 'semantic-ui-react';
+import type { LabelInputType } from '../labels/types';
 
-const allowedCharacters = /^[a-z][a-z0-9._-]*$/i;
+const allowedCharactersForLabelKey = /^[a-z0-9._-]*$/i;
+const allowedCharactersForLabelValue = /^[^\p{C}"]*$/u;
 const maxInputLength = 256;
 
-// TODO: Add proper typing for onChange function
-function useLabelInput(onChange: (value: any) => void, { allowAnyValue = false, initialValue = '' }) {
+function removeControlCharacters(value: string) {
+    return value.replace(/\p{C}/gu, '');
+}
+
+function formatNewValue(type: 'key' | 'value', newValue: string) {
+    const truncatedNewValue = newValue.substr(0, maxInputLength);
+    return type === 'key' ? truncatedNewValue.toLowerCase() : removeControlCharacters(truncatedNewValue);
+}
+
+function useLabelInput(
+    onChange: (value: string) => void,
+    type: LabelInputType,
+    { initialValue } = { initialValue: '' }
```

Comment:
Minor:

Technically, doing the following allows for more:

```suggestion
    { initialValue = '' } = { }
```

and is shorter :smile: If there were more properties in the object, the existing solution would make them all required, whereas the one I suggested makes the `initialValue` property optional

Reply:
Right, changed 👍🏻